modules/sdk-lib-mpc/src/tss/ecdsa-dkls/util.ts

```diff
@@ -79,3 +82,174 @@ export function verifyAndConvertDklsSignature(
     dklsSignature.S
   ).toString('hex')}${delimeter}${truePub}`;
 }
+
+export async function executeTillRound(
+  round: number,
+  party1Dsg: Dsg,
+  party2Dsg: Dsg
+): Promise<DeserializedMessages[] | DeserializedDklsSignature> {
+  if (round < 1 || round > 5) {
+    throw Error('Invalid round number');
+  }
+  const party1Round1Message = await party1Dsg.init();
+  const party2Round1Message = await party2Dsg.init();
+
+  const party2Round2Messages = party2Dsg.handleIncomingMessages({
+    p2pMessages: [],
+    broadcastMessages: [party1Round1Message],
+  });
+  const party1Round2Messages = party1Dsg.handleIncomingMessages({
+    p2pMessages: [],
+    broadcastMessages: [party2Round1Message],
+  });
+  if (round === 2) return [party1Round2Messages, party2Round2Messages];
+
+  const party1Round3Messages = party1Dsg.handleIncomingMessages({
+    p2pMessages: party2Round2Messages.p2pMessages,
+    broadcastMessages: [],
+  });
+  const party2Round3Messages = party2Dsg.handleIncomingMessages({
+    p2pMessages: party1Round2Messages.p2pMessages,
+    broadcastMessages: [],
+  });
+  if (round === 3) return [party1Round3Messages, party2Round3Messages];
+
+  const party2Round4Messages = party2Dsg.handleIncomingMessages({
+    p2pMessages: party1Round3Messages.p2pMessages,
+    broadcastMessages: [],
+  });
+  const party1Round4Messages = party1Dsg.handleIncomingMessages({
+    p2pMessages: party2Round3Messages.p2pMessages,
+    broadcastMessages: [],
+  });
+  if (round === 4) return [party1Round4Messages, party2Round4Messages];
+
+  party1Dsg.handleIncomingMessages({
+    p2pMessages: [],
+    broadcastMessages: party2Round4Messages.broadcastMessages,
+  });
+  party2Dsg.handleIncomingMessages({
+    p2pMessages: [],
+    broadcastMessages: party1Round4Messages.broadcastMessages,
+  });
+  assert(Buffer.from(party1Dsg.signature.R).toString('hex') === Buffer.from(party2Dsg.signature.R).toString('hex'));
+  assert(Buffer.from(party1Dsg.signature.S).toString('hex') === Buffer.from(party2Dsg.signature.S).toString('hex'));
+  return party1Dsg.signature;
+}
+
+export async function generateDKGKeyShares(
+  retrofitDataA?: RetrofitData,
+  retrofitDataB?: RetrofitData,
+  retrofitDataC?: RetrofitData
+): Promise<[Dkg, Dkg, Dkg]> {
+  const user = new Dkg(3, 2, 0, retrofitDataA);
+  const backup = new Dkg(3, 2, 1, retrofitDataB);
+  const bitgo = new Dkg(3, 2, 2, retrofitDataC);
+  const userRound1Message = await user.initDkg();
+  const backupRound1Message = await backup.initDkg();
+  const bitgoRound1Message = await bitgo.initDkg();
+  const userRound2Messages = user.handleIncomingMessages({
+    p2pMessages: [],
+    broadcastMessages: [bitgoRound1Message, backupRound1Message],
+  });
+  const backupRound2Messages = backup.handleIncomingMessages({
```

Comment:
nit: sdk-lib-mpc should be agnostic to bitgo modeling, so we should remove the user/backup/bitgo naming scheme where possible.